test/cpp_extensions/setup.py

```diff
@@ -1,13 +1,21 @@
-from setuptools import setup, Extension
-import torch.utils.cpp_extension
+import torch.cuda
+from setuptools import setup
+from torch.utils.cpp_extension import CppExtension, CUDAExtension
 
 ext_modules = [
-    Extension(
+    CppExtension(
         'torch_test_cpp_extensions', ['extension.cpp'],
-        include_dirs=torch.utils.cpp_extension.include_paths(),
-        language='c++'),
+        extra_compile_args=['-g']),
 ]
 
+if torch.cuda.is_available():
+    extension = CUDAExtension(
+        'torch_test_cuda_extension',
+        ['cuda_extension.cpp', 'cuda_extension_kernel.cu'],
+        extra_compile_args={'cxx': ['-g'],
+                            'nvcc': ['-arch=sm_30']})
```

Comment:
Are you sure you want this hardcoded here?

Reply:
Yeah I should change I'll change it to `-O2` or something -- it's just to test this dictionary `extra_compile_args` behavior.